src/main/java/com/epam/aidial/core/config/CoreResourceAuthSettings.java

```diff
@@ -0,0 +1,59 @@
+package com.epam.aidial.core.config;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+import java.util.List;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Accessors(chain = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class CoreResourceAuthSettings {
+
+    @NotNull(message = "Authentication type must be defined")
+    @JsonAlias({"authenticationType", "authentication_type"})
+    @Builder.Default
+    private CoreAuthenticationType authenticationType = CoreAuthenticationType.NONE;
+
+    @JsonAlias({"clientId", "client_id"})
+    private String clientId;
+
+    @JsonAlias({"clientSecret", "client_secret"})
+    private String clientSecret;
```

Comment:
clientSecret is secret value. should be:
a) controlled by `include_secrets` options in export config functionality
b) exported in secret storage in scheduled core config export